src/lib/journey/journey.store.ts

```diff
@@ -228,11 +277,25 @@ export function initialize(initOptions?: StepOptions): JourneyStore {
     }
   }
 
+  async function pop() {
+    reset();
+    const updatedStack = await stack.pop();
+    const currentJourney = updatedStack[updatedStack.length - 1];
+    await start(currentJourney);
+  }
+
+  async function push(newOptions: StepOptions) {
+    reset();
+    await stack.push(newOptions);
+    await start(newOptions);
+  }
+
```

Comment:
This adds the stack's methods to the same `journey` store that is already being used. This prevents the "things" we need to pass around to a minimum.